File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    s, e = 0, len(people) - 1
    people.sort()
    while s <= e:
        if people[s] + people[e] <= limit:
            answer += 1
            s += 1
            e -= 1
        elif people[s] + people[e] > limit:
            e -= 1
            answer += 1

    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    static public int solution(int[] people, int limit) {
        int answer = 0;
        int s = 0;
        int e = people.length - 1;
        Arrays.sort(people);
        while (s <= e) {
            if (people[s] + people[e] <= limit) {
                s += 1;
                e -= 1;
                answer += 1;
            } else if (people[s] + people[e] > limit) {
                answer += 1;
                e -= 1;
            }
        }
        return answer;
    }
}